webapp/src/main/java/com/box/l10n/mojito/service/tm/TMTextUnitStatisticService.java

```diff
@@ -0,0 +1,216 @@
+        TextUnitDTO foundTextUnitDTOByMD5 = textUnitDTOsByMD5.get(statisticTextUnitMD5);
+
+        if (foundTextUnitDTOByMD5 != null) {
+            return ImmutableList.of(foundTextUnitDTOByMD5);
+        }
+
+        // Only attempt a fallback match by name and content if a comment isn't available in the statistic object
+        // as different comment values shouldn't result in a match otherwise.
+        if (StringUtils.isEmpty(textUnitStatistic.getComment())) {
+            List<TextUnitDTO> foundTextUnitDTOsByNameAndContent = textUnitDTOsByNameAndContent.get(
+                    new ImmutablePair<>(textUnitStatistic.getName(), textUnitStatistic.getContent())
+            );
+
+            if (foundTextUnitDTOsByNameAndContent != null && foundTextUnitDTOsByNameAndContent.size() > 0) {
+                return ImmutableList.copyOf(foundTextUnitDTOsByNameAndContent);
+            }
+        }
+
+        return null;
+    }
+
+    private void logUnmatchedStatistics(Asset asset, ImportTextUnitStatisticsBody textUnitStatistic) {
+        String message = String.format(
+                "No equivalent text unit found, skip import statistics for text unit with name: %1$s for asset with id: %2$s & path: %3$s",
+                textUnitStatistic.getName(),
+                asset.getId(),
+                asset.getPath());
+
+        logger.warn(message);
+
+        Tags tags = Tags.of("assetId", String.valueOf(asset.getId()),
+                "assetPath", asset.getPath());
+
+        meterRegistry.summary("services.TMTextUnitStatisticService.import_unmatched_statistics", tags);
+    }
+
+    private Map<ImportTextUnitStatisticsBody, TMTextUnit> getStatisticsToTextUnitMap(Map<ImportTextUnitStatisticsBody, ImmutableList<TextUnitDTO>> statisticToTextUnitDTOMap, List<Long> textUnitIds) {
+        List<TMTextUnit> textUnits = new ArrayList<>();
+        Iterables.partition(textUnitIds, batchSize).forEach(
+                textUnitIdBatch -> textUnits.addAll(tmTextUnitRepository.findByIdInAndEagerFetchStatistics(textUnitIdBatch))
+        );
+
+        Map<Long, TMTextUnit> idTextUnitMap = textUnits.stream().collect(
+                ImmutableMap.toImmutableMap(BaseEntity::getId, Function.identity())
+        );
+
+        Map<ImportTextUnitStatisticsBody, TMTextUnit> statisticsTMTextUnitMap = new HashMap<>();
+        statisticToTextUnitDTOMap.forEach((textUnitStatisticsBody, textUnitDTOs) -> {
+            textUnitDTOs.forEach(textUnitDTO -> {
+                statisticsTMTextUnitMap.put(textUnitStatisticsBody, idTextUnitMap.get(textUnitDTO.getTmTextUnitId()));
+            });
+        });
+
+        return statisticsTMTextUnitMap;
+    }
+
+    private void uploadStatistics(Map <ImportTextUnitStatisticsBody, TMTextUnit> statisticsToTextUnitsMap) {
+        List<TMTextUnitStatistic> updatedStatistics = new ArrayList<>();
+
+        statisticsToTextUnitsMap.forEach((textUnitStatistic, tmTextUnit) -> {
```

Comment:
`stream().map((textUnitStatistic, tmTextUnit) -> {
Write Preview
 
Attach files by dragging & dropping, selecting or pasting them.
Start a review
Add single comment
Cancel
            logger.debug("Add statistics for text unit with name: {}.", textUnitStatistic.getName());

            TMTextUnitStatistic statistic = tmTextUnit.getStatistic();

            if (statistic == null) {
                statistic = new TMTextUnitStatistic();
                statistic.setTMTextUnit(tmTextUnit);
            }

            statistic.setLastDayUsageCount(textUnitStatistic.getLastDayEstimatedVolume());
            statistic.setLastPeriodUsageCount(textUnitStatistic.getLastPeriodEstimatedVolume());
            statistic.setLastSeenDate(textUnitStatistic.getLastSeenDate());

return stastic;
        }).collector(toList());

Reply:
Changed in the latest iteration.